cmstaskenv/cmsMake.py

```diff
@@ -466,17 +466,44 @@ def make_output(n, assume=None):
                                 stream=sys.stderr, bold=True),
             file=sys.stderr
         )
-        with io.open(os.path.join(input_dir,
-                                  'input%d.txt' % (n)), 'rb') as fin:
-            with io.open(os.path.join(output_dir,
-                                      'output%d.txt' % (n)), 'wb') as fout:
-                if task_type != ['Communication', '']:
-                    call(base_dir, [sol_exe], stdin=fin, stdout=fout)
-                    move_cursor(directions.UP, erase=True, stream=sys.stderr)
-                # If the task of of type Communication, then there is
-                # nothing to put in the output files
-                else:
-                    pass
+
+        temp_dir = tempfile.mkdtemp()
+        use_stdin = yaml_conf.get("infile") in {None, ""}
+        use_stdout = yaml_conf.get("outfile") in {None, ""}
+
+        # Names of the actual source and destination.
+        infile = os.path.join(input_dir, 'input%d.txt' % (n))
+        outfile = os.path.join(output_dir, 'output%d.txt' % (n))
+
+        # Names of the input and output in temp directory.
+        copied_infile = os.path.join(
+            temp_dir,
+            "input.txt" if use_stdin else yaml_conf.get("infile"))
+        copied_outfile = os.path.join(
+            temp_dir,
+            "output.txt" if use_stdout else yaml_conf.get("outfile"))
+
+        shutil.copy(infile, copied_infile)
+        fin = io.open(copied_infile, "rb") if use_stdin else None
+        fout = io.open(copied_outfile, 'wb') if use_stdout else None
```

Comment:
The 2 lines above should use the `with ... as fin:` and `with ... as fout:` syntax